test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SigningUtilityTests.cs

```diff
@@ -36,24 +36,37 @@ public void Verify_WithValidInput_DoesNotThrow()
         }
 
         [CIOnlyFact]
-        public async Task SignAsync_AddsPackageSignatureAsync()
+        public async Task SignAsync_AddsPackageAuthorSignatureAsync()
         {
             using (var test = new Test(_testFixture.TrustedTestCertificate.Source.Cert))
             {
-                await SigningUtility.SignAsync(test.Options, test.Request, CancellationToken.None);
+                await SigningUtility.SignAsync(test.Options, test.AuthorRequest, CancellationToken.None);
 
                 var isSigned = await IsSignedAsync(test.Options.OutputFilePath);
 
                 Assert.True(isSigned);
             }
         }
+        [CIOnlyFact]
+        public async Task SignAsync_AddsPackageRepositorySignatureAsync()
```

Comment:
This is great, but now that we are creating multiple kinds of primary signatures we should validate that the result is properly repository signed.
```
using (var package = new PackageArchiveReader(File.OpenRead(test.Options.OutputFilePath)))
{
    var primarySignature = await package.GetPrimarySignatureAsync(CancellationToken.None);

    Assert.IsType<RepositoryPrimarySignature>(primarySignature);
}
```
This replaces the "is signed" test already in the test.

The [test above](https://github.com/NuGet/NuGet.Client/pull/2064/files#diff-cc2b51ac0f5e1bae62f3aee4c89a911cR39) should do the same but assert:

    Assert.IsType<AuthorPrimarySignature>(primarySignature);